### PR TITLE
[Core] lazy import autoscaler + don't import opentelemetry unless setup hook

### PR DESCRIPTION
**Pull request by rkooo567**
… is configured

## Why are these changes needed?

This will improve startup time almost 2X and reduce memory usage by 2X. If it is combined with numpy lazy import, it will improve everything 3X. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

File: python/ray/__init__.py [hunk 1]
```diff
@@ -256,12 +255,13 @@ def __getattr__(self, attr):
     # TODO(Clark): Remove this one we drop Python 3.6 support.
     from ray import data  # noqa: F401
     from ray import workflow  # noqa: F401
+    from ray import autoscaler  # noqa: E402,F401
```
Comment (cadedaniel) [inline python/ray/__init__.py:258]: nit: add a comment why this is here?

File: python/ray/_private/worker.py [hunk 2]
```diff
@@ -2285,7 +2285,7 @@ def connect(
         b"tracing_startup_hook", ray_constants.KV_NAMESPACE_TRACING
     )
     if tracing_hook_val is not None:
-        ray.util.tracing.tracing_helper._global_is_tracing_enabled = True
+        ray.util.tracing.tracing_helper._enbale_tracing()
```
Comment (cadedaniel) [inline python/ray/_private/worker.py:2288]: typo here? `enbale`

File: python/ray/util/tracing/tracing_helper.py [hunk 3]
```diff
@@ -84,11 +84,22 @@ def _try_import(self, module):
...
+def _is_tracing_enabled() -> bool:
+    """Checks environment variable feature flag to see if tracing is turned on.
+    Tracing is off by default."""
+    return _global_is_tracing_enabled
+
+
+def _enbale_tracing():
+    global _global_is_tracing_enabled
+    _global_is_tracing_enabled = True
+    _opentelemetry = _OpenTelemetryProxy()
```
Comment (cadedaniel) [inline python/ray/util/tracing/tracing_helper.py:101]: for my knowledge what does this do? does all our worker telemetry flow through this?

**Review by cadedaniel — APPROVED** [review]
nice find :) only nits

**Comment by cadedaniel** [comment]
looks like some related test failures as well.

File: python/ray/util/tracing/tracing_helper.py (hunk 3 above)
Comment (rkooo567) [reply python/ray/util/tracing/tracing_helper.py:101]: it's an experimental that we merged long time ago for distributed tracing. I believe almost no one uses this feature now, but it has really tight integration to core

File: python/ray/__init__.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/__init__.py:258]: sure. it is to enable ray.autoscaler.sdk.request_resources I believe

**Comment by rkooo567** [comment]
Most of new code changes is to fix the package resolution error from tests that happen to work.

**Pull request merged**